File: src/lib/software/microsoft_teams.py
from src.lib.software import BaseSoftware

class MicrosoftTeams(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'MSTeamsSetup.exe'
        self.silent_install_args = ['--silent']

    def resolve_download_url(self):
        self.url_resolved.emit('https://go.microsoft.com/fwlink/?linkid=2281613&clcid=0x409&culture=en-us&country=us')

File: packages/app/src/services/software/definitions/microsoftTeams.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'microsoft-teams';
	public name = 'Microsoft Teams';
	public category = [SoftwareCategory.Social];
	public downloadName = 'MSTeamsSetup.exe';
	public isArchive = false;
	public shouldCacheUrl = false;
	public requiresAdmin = false;
	public deprecated = false;
	public icon = 'microsoft-teams.png';
	public homepage = 'https://microsoft.com/en-us/microsoft-teams';

	public async resolveDownloadUrl() {
		return ok('https://go.microsoft.com/fwlink/?linkid=2281613&clcid=0x409&culture=en-us&country=us');
	}
}
